src/mapd-con-es6.js

```diff
@@ -1005,6 +805,16 @@ class MapdCon {
       })
     )
 
+  detectColumnTypes(fileName, copyParams, callback) {
+    const thriftCopyParams = helpers.convertObjectToThriftCopyParams(copyParams)
+    this._client[0].detect_column_types(
```

Comment:
So, I say this with the knowledge that I may have been the last one to edit this function, but is there a reason we're calling the `this._client[0].detect_column_types` function directly rather than wrapping it like we do with most other functions here.

Reply:
Good question. I can't see any reason we couldn't wrap this one too. I'll do that real quick and spot-test it in the importer.

Reply:
I found out why we can't wrap it - we need to mutate `this.importerRowDesc` with the result that returns, for the benefit of subsequent `createTable` and `importTable` calls. A horrible, stateful pattern we should fix, but a little beyond the scope of this.